common/src/services/vaultTimeout.service.ts

```diff
@@ -141,6 +138,25 @@ export class VaultTimeoutService implements VaultTimeoutServiceAbstraction {
         return await this.storageService.get<boolean>(ConstantsService.biometricUnlockKey);
     }
 
+    async getVaultTimeout(): Promise<number> {
+        const vaultTimeout = await this.storageService.get<number>(ConstantsService.vaultTimeoutKey);
+
+        if (await this.policyService.policyAppliesToUser(PolicyType.MaximumVaultTimeout)) {
+            const policy = await this.policyService.getAll(PolicyType.MaximumVaultTimeout);
+            // Remove negative values, and ensure it's smaller than maximum allowed value according to policy
+            const timeout = Math.max(0, Math.min(vaultTimeout, policy[0].data.minutes));
+
+            // We really shouldn't need to set the value here, but multiple services relies on this value being correct.
+            if (vaultTimeout != timeout) {
+                await this.storageService.save(ConstantsService.vaultTimeoutKey, timeout);
+            }
```

Comment:
I really don't like this, but the dependency chain for `VaultTimeoutService` would need to be reversed to clean this up. Which feels out of scope for this issue.